BUILD.md

```diff
@@ -55,3 +53,9 @@ System package managers, and language package managers have to build on multiple
 By defaulting to `ON` we cause issues for package managers since there is no standard way to disable warnings until CMake 3.24
 
 Add `-D VUL_WERROR=ON` to your workflow. Or use the `dev` preset shown below which will also enabling warnings as errors.
+
+### clang-tidy
+
+By default clang-tidy checking is disabled by default.
```

Comment:
Why is it disabled by default? It seems much better for me to hit issues on my system than on C.I., consume more C.I. resource and have to iterate, taking more development time. This is especially true that I don't see how the developer would even now what the rules are.

I would argu the same with VUL_WERROR being OFF by default.

I think we need to be more consious when we add new checks how this is going to affect the developers productivity.

Reply:
It's off by default to avoid issues with package managers, machines that don't have clang-tidy installed by default, FetchContent/add_subdirectory users, etc.

Having it on by default is more complicated than having it on. That's just the way it is with open source projects.